rstudio-entrypoint.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# On Linux set rstudio user id to same as id of host if rstudio user id not already the same
+if [ "$HOSTPLATFORM" = "linux" -a "$(id -u rstudio)" != "$HOSTUID" ]; then
+  usermod -u $HOSTUID rstudio
+fi
+
+# Check for 1 .Rproj file
+if [ $(find /workspace -type f -name "*.Rproj" | wc -w) -eq 1 ]; then
+
+  # Copy in Git user.name and user.email from copied local-gitconfig from additionally mounted volume
+  if test -f /home/rstudio/local-gitconfig; then
+    grep -e "\[user\]" -e "name = *" -e "email = *" /home/rstudio/local-gitconfig >> /home/rstudio/.gitconfig
+  fi
+
+  # Avoid Git error: fatal detected dubious ownership of repository if using Git in container
+  # Without this the Git pane fails to open when RStudio project opened
+  echo -e "[safe]\n\tdirectory = \"*\"" >> /home/rstudio/.gitconfig
+
+  # Open RStudio project on opening RStudio Server session
+  echo 'setHook("rstudio.sessionInit", function(newSession) { if (newSession && is.null(rstudioapi::getActiveProject())) rstudioapi::openProject(paste0("/workspace/", list.files(pattern = "Rproj"))) }, action = "append")' >> /home/rstudio/.Rprofile
+fi
+
+# Set file line endings as crlf if docker run from Windows
+if [ "$HOSTPLATFORM" = "win32" ]; then
+  echo -e "{\n\t\"line_ending_conversion\": \"windows\"\n}" >> /etc/rstudio/rstudio-prefs.json
+fi
+
+# Start RStudio Server session in foreground
+# Hence don't use `rstudio-server start` which runs in background
+# and attempt to capture std out and err to a metadata log file
+mkdir -p /workspace/metadata
+exec /usr/lib/rstudio-server/bin/rserver --server-daemonize 0 >> /workspace/metadata/rstudio.log 2>&1
```

Comment:
This isn't right, but as discussed, I will investigate.

Reply:
I amended the `>>` to `>` but I can't improve on this atm.

Reply:
As discussed, it turned out, when stderr was not redirected, the messages I saw in Terminal were only printed because rstudio-server detected stderr was going to a Terminal.

<https://github.com/rstudio/rstudio/blob/8ce4df540d346d9fd64507d4b4af16c66f3ef10b/src/cpp/core/system/System.cpp#L242-L251>

So we've left the redirection to rstudio.log in case anything useful shows up, but also to prevent the stderr informational messages about logging being shown in the Terminal. I've added comments in *rstudio-entrypoint.sh* that there are two other additional log files recorded in the container at `/var/log/rstudio/rstudio-server/rserver.log` and `/home/rstudio/.local/share/rstudio/log/rsession-rstudio.log`. For these two log files I have increased the `log-level` to `info` (`/etc/rstudio/logging.conf` amended in Dockerfile).